Tools/scripts/validate_board_hwdef.py

```diff
@@ -0,0 +1,353 @@
+            cwd=repo_root,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+        submodule_paths = result.stdout.splitlines()
+
+        for submodule_path in submodule_paths:
+            submodule_dir = repo_root / submodule_path
+            if not submodule_dir.exists():
+                continue
+
+            for relative_path in get_git_files(submodule_dir):
+                if not should_check_file(relative_path):
+                    continue
+
+                filepath = submodule_dir / relative_path
+                try:
+                    content = filepath.read_text(encoding='utf-8', errors='ignore')
+                    content_parts.append(content)
+                except Exception:
+                    # Skip files that can't be read
+                    pass
+    except subprocess.CalledProcessError:
+        pass
+
+    # Join all content with newlines and cache it
+    _codebase_content_cache = '\n'.join(content_parts)
+    return _codebase_content_cache
+
+
+def validate_board_worker(args: Tuple) -> Tuple[str, Optional[Dict[str, str]], Optional[str]]:
+    """
+    Worker function for parallel board validation.
+
+    Args:
+        args: Tuple of (board, board_num, total_boards, verbose)
+
+    Returns:
+        Tuple of (board_name, unused_defines_dict, error_message)
+    """
+    board, board_num, total_boards, verbose = args
+
+    # Build codebase content once per worker process (cached)
+    codebase_content = build_codebase_content()
+
+    # Import here to avoid issues with multiprocessing
+    board_list = BoardList()
+
+    try:
+        # Get the HWDef instance for this board
+        hwdef_instance = board_list.get_HWDef_instance(board)
+
+        if hwdef_instance is None:
+            # SITL board - no hwdef file
+            return (board.name, None, None)
+
+        # Process the hwdef files to populate the defines dictionary
+        hwdef_instance.process_hwdefs()
+
+        # Get defines from the hwdef instance
+        defines = hwdef_instance.defines
+
+        # Check each define
+        unused = {}
+        for define_name, line in defines.items():
+            # First check if this define is used by other defines in the same hwdef
+            used_in_hwdef = False
+            for other_name, other_value in defines.items():
+                if other_name != define_name and other_value is not None:
+                    if define_name in other_value:
+                        used_in_hwdef = True
+                        break
+
+            # If not used in hwdef, check if used in codebase
+            if not used_in_hwdef and not check_define_in_codebase(define_name, codebase_content):
+                unused[define_name] = line
+
+        return (board.name, unused if unused else None, None)
+
+    except Exception as e:
+        error_msg = f"Error validating board {board.name}: {e}"
+        return (board.name, None, error_msg)
```

Comment:
Throwing away the traceback like this is very annoying for debugging. It should just die so it can be fixed.